plugins/kubernetes/src/hooks/auth.test.ts

```diff
@@ -0,0 +1,125 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import { generateAuth } from './auth';
+
+jest.mock('@backstage/core-plugin-api');
+
+const entity = {
+  metadata: {
+    name: 'some-entity',
+  },
+} as Entity;
+
+const entityWithAuthToken = {
+  auth: {
+    google: 'some-token',
+  },
+  entity,
+};
+
+const getClustersResponse = [
+  {
+    name: 'cluster-a',
+    authProvider: 'google',
+  },
+  {
+    name: 'cluster-b',
+    authProvider: 'authprovider2',
+  },
+];
+
+describe('generateAuth', () => {
+  const mockGetClusters = jest.fn();
+  const mockDecorateRequestBodyForAuth = jest.fn();
+
+  const expectMocksCalledCorrectly = (numOfCalls: number = 1) => {
+    expect(mockGetClusters).toHaveBeenCalledTimes(numOfCalls);
+    expect(mockGetClusters).toHaveBeenLastCalledWith();
+    expect(mockDecorateRequestBodyForAuth).toHaveBeenCalledTimes(
+      numOfCalls * 2,
+    );
+    expect(mockDecorateRequestBodyForAuth).toHaveBeenCalledWith('google', {
+      entity,
+    });
+    expect(mockDecorateRequestBodyForAuth).toHaveBeenCalledWith(
+      'authprovider2',
+      entityWithAuthToken,
+    );
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+  it('should return auth', async () => {
+    const result = await generateAuth(
+      entity,
+      {
+        getClusters: mockGetClusters.mockResolvedValue(getClustersResponse),
+      } as any,
+      {
+        decorateRequestBodyForAuth:
+          mockDecorateRequestBodyForAuth.mockResolvedValue(entityWithAuthToken),
+      } as any,
+    );
+
+    expect(result).toStrictEqual({
+      google: 'some-token',
+    });
+    expectMocksCalledCorrectly();
+  });
+
+  it('should return error when getClusters throws', async () => {
+
+
+    expect(async () =>  await generateAuth(
+      entity,
+      {
+        getClusters: mockGetClusters.mockRejectedValue('some-error'),
+      } as any,
+      {
+        decorateRequestBodyForAuth: mockDecorateRequestBodyForAuth,
+      } as any,
+    ))
+    .rejects
+    .toThrow()
+
+    expect(mockGetClusters).toHaveBeenCalledTimes(1);
+    expect(mockGetClusters).toHaveBeenLastCalledWith();
+    expect(mockDecorateRequestBodyForAuth).toHaveBeenCalledTimes(0);
+  });
+  it('should return error when decorateRequestBodyForAuth throws', async () => {
+    expect(async () =>  await generateAuth(
+      entity,
+      {
+        getClusters: mockGetClusters.mockRejectedValue('some-error'),
+      } as any,
+      {
+        decorateRequestBodyForAuth: mockDecorateRequestBodyForAuth,
+      } as any,
+    ))
+    .rejects
+    .toThrow()
```

Comment:
@Rugvip any idea why these tests don't work?

Reply:
Yep you need to `await` the expect up at line 106

Reply:
I tried that, then it says `Received function did not throw`